syncopy/tests/test_connectivity.py

```diff
@@ -0,0 +1,584 @@
+# Decorator to decide whether or not to run memory-intensive tests
+availMem = psutil.virtual_memory().total
+minRAM = 5
+skip_low_mem = pytest.mark.skipif(availMem < minRAM * 1024**3, reason=f"less than {minRAM}GB RAM available")
+
+
+class TestGranger:
+
+    nTrials = 100
+    nChannels = 5
+    nSamples = 1000
+    fs = 200
+
+    # -- Create a somewhat intricate
+    # -- network of AR(2) processes
+
+    # the adjacency matrix
+    # encodes coupling strength directly
+    AdjMat = np.zeros((nChannels, nChannels))
+    AdjMat[0, 4] = 0.15
+    AdjMat[3, 4] = 0.15
+    AdjMat[3, 2] = 0.25
+    AdjMat[1, 0] = 0.25
+
+    print(AdjMat)
+    # channel indices of coupling
+    # a number other than 0 at AdjMat(i,j)
+    # means coupling from i->j
+    cpl_idx = np.where(AdjMat)
+    nocpl_idx = np.where(AdjMat == 0)
+
+    trls = []
+    for _ in range(nTrials):
+        # defaults AR(2) parameters yield 40Hz peak
+        trls.append(synth_data.AR2_network(AdjMat, nSamples=nSamples))
+
+    # create syncopy data instance
+    data = AnalogData(trls, samplerate=fs)
+    time_span = [-1, nSamples / fs - 1]   # -1s offset
+    foi = np.arange(5, 75)   # in Hz
+
+    def test_gr_solution(self, **kwargs):
+
+        Gcaus = connectivity(self.data, method='granger', taper='dpss',
+                             tapsmofrq=3, foi=self.foi, **kwargs)
+
+        # print("peak \t Aij \t peak-frq \t ij")
+        # check all channel combinations with coupling
+        for i, j in zip(*self.cpl_idx):
+            peak = Gcaus.data[0, :, i, j].max()
+            peak_frq = Gcaus.freq[Gcaus.data[0, :, i, j].argmax()]
+            cval = self.AdjMat[i, j]
+
+            dbg_str = f"{peak:.2f}\t{self.AdjMat[i,j]:.2f}\t {peak_frq:.2f}\t"
+            print(dbg_str, f'\t {i}', f' {j}')
+
+            # test for directional coupling
+            # at the right frequency range
+            assert peak >= cval
+            assert 35 < peak_frq < 45
+
+            # only plot with defaults
+            if len(kwargs) == 0:
+                plot_Granger(Gcaus, i, j)
+                ppl.legend()
+
+    def test_gr_selections(self):
+
+        # trial, channel and toi selections
+        selections = mk_selection_comb(self.nTrials,
+                                       self.nChannels,
+                                       *self.time_span)
+
+        for sel_dct in selections:
+
+            Gcaus = connectivity(self.data, method='granger', select=sel_dct)
+
+            # check here just for finiteness and positivity
+            assert np.all(np.isfinite(Gcaus.data))
+            assert np.all(Gcaus.data[0, ...] >= -1e-10)
+
+    def test_gr_foi(self):
+
+        call = lambda foi, foilim: connectivity(self.data,
+                                                method='granger',
+                                                foi=foi,
+                                                foilim=foilim)
+
```

Comment:
I'm probably missing the obvious here, but why is `foi`/`foilim` tested separately and not included in `mk_selection_comb`?

Reply:
Because they are not part of a selection dict for `AnalogData`